lib/cosmos/interfaces/protocols/stream_protocol.rb

```diff
@@ -38,50 +36,24 @@ class StreamProtocol
     #   that will be searched for in the raw {Stream}. Bytes encountered before
     #   this pattern is found are discarded.
     # @param fill_fields [Boolean] Fill any required fields when writing packets
-    def initialize(discard_leading_bytes = 0, sync_pattern = nil, fill_fields = false)
+    def configure_stream_protocol(discard_leading_bytes = 0, sync_pattern = nil, fill_fields = false)
       @discard_leading_bytes = discard_leading_bytes.to_i
       @sync_pattern = ConfigParser.handle_nil(sync_pattern)
       @sync_pattern = @sync_pattern.hex_to_byte_string if @sync_pattern
       @fill_fields = ConfigParser.handle_true_false(fill_fields)
-
-      @stream = nil
       @data = ''
-      @bytes_read = 0
-      @bytes_written = 0
-
-      @interface = Interface.new
-      @write_mutex = Mutex.new
-    end
-
-    # @param interface [Interface] Sets the higher level interface which is
-    #   using this StreamProtocol.
-    def interface=(interface)
-      @interface = interface
     end
 
-    # @param stream [Stream] The stream this stream protocol should read and
-    #   write to
-    def connect(stream)
+    # Clears the data attribute and sets the data encoding
+    def connect
+      super()
       @data = ''
       @data.force_encoding('ASCII-8BIT')
-      @stream = stream
-      @stream.connect
```

Comment:
Connecting the stream is done at a higher level. All we concern ourselves with is the protocol so we clear @data.